Roll-Play/roll-play-api
# feat(ci): add CI workflow

## Pull Request Description

### Changes

In this pull request, we are adding GitHub Actions to automate the build process and run tests for our Go program. This will help us ensure code quality and catch any errors early in the development process.

### Proposed Changes

- Set up a GitHub Actions workflow (`/.github/workflows/ci.yml`) that triggers on each pull request.
- Configure the workflow to:
  - Build the Go program using Go's standard build commands.
  - Run tests using `go test` to check for code correctness.
  - Provide feedback on the PR status page about the build and test results.

### Why is this Important?

Automating the build and testing process using GitHub Actions helps us maintain code quality and catch potential issues before they reach production. It also ensures that new contributions are well-tested and aligned with our coding standards.

### Testing Instructions

1. Open the PR and review the changes.
2. Observe the GitHub Actions workflow status checks to see if the build and tests pass.
3. If there are any issues, review the workflow logs for more details.
4. Once all checks pass, the PR is ready for review and merging.
